code/modules/mob/living/simple_animal/friendly/spiderbot.dm

```diff
@@ -119,25 +119,34 @@
 			return
 	else if(O.GetID())
 		if (!mmi)
-			to_chat(user, "<span class='danger'>There's no reason to swipe your ID - \the [src] has no brain to remove.</span>")
+			to_chat(user, "<span class='danger'>There's no reason to swipe your ID - \the [src] has no brain.</span>")
 			return 0
 
 		var/obj/item/card/id/id_card = O.GetID()
 
 		if(!istype(id_card))
 			return 0
 
-		if(access_robotics in id_card.access)
-			to_chat(user, "<span class='notice'>You swipe your access card and pop the brain out of \the [src].</span>")
-			eject_brain()
-			if(held_item)
-				held_item.forceMove(src.loc)
-				held_item = null
-			return 1
-		else
-			to_chat(user, "<span class='danger'>You swipe your card with no effect.</span>")
-			return 0
+		var/choice = input(user, "Would you like to eject the brain or sync access?", "Swipe Mode") as null|anything in list("Eject", "Sync")
+		if(!choice)
+			return
 
+		switch(choice)
```

Comment:
Before doing that you need to check if they are still within range / is capable of doing that.
(As someone might swipe the id and get the prompt, then wait for x minutes, move away and select a option from across the station.)